index.html

```diff
@@ -11,32 +11,201 @@
+			const searchButton = document.querySelector('#search-button');
+
+			searchField.addEventListener('keydown', (event) => {
+				if (event.keyCode === 13) {
+					event.preventDefault();
+					const request = setupRequest();
+					request.send();
+				}
+			});
+
+			searchButton.addEventListener('click', (event) => {
+				event.preventDefault();
+
+				const request = setupRequest();
+				request.send();
+			});
+
+
+			function setupRequest () {
+				const request = new XMLHttpRequest();
+
+				request.open('GET', 'https://api.nytimes.com/svc/topstories/v2/science.json?api-key=Gwxln5M3geWlhR6UE0TY1FUWKSG3wCil');
+
+				request.onload = handleResponse;
+
+				request.onerror = function() {
+					console.error(this);
+				};
+
+				return request;
+			}
+
+			function handleResponse () {
 
-			request.onload = function() {
 				if (this.status >= 200 && this.status < 400) {
-					console.log(JSON.parse(this.response));
+					const parsedResponse = JSON.parse(this.response);
+					if (parsedResponse.results && Array.isArray(parsedResponse.results)) {
+
+						// remove existing children from results
+						while (resultsList.firstChild) {
+							resultsList.removeChild(resultsList.firstChild);
+						}
+
+						// split search input into array so order doesn't matter
+						const searchInput = searchField.value.split(' ');
+						let matchedResults = [];
+
+						parsedResponse.results.forEach((item) => {
+							// combine all searchable fields into one block of text
+							const searchableText = `${item.title} ${item.byline} ${item.section}`.toLowerCase();
+							let match = false;
+							let relevance = 0;
+
+							// ignore blank spaces in search input
+							if (searchInput.length > 0 && searchInput[0] !== '') {
+								searchInput.forEach((searchWord) => {
+
+									// basic regex search pattern so we match parts of words as well
+									// e.g. chan would work for change
+									const pattern = new RegExp(searchWord, 'g');
+									const matches = searchableText.match(pattern);
+
+									if (searchWord !== '' && matches && matches.length > 0) {
+										match = true;
+
+										// basic relevance based on number of matches we get per searchable text
+										relevance += matches.length;
+									}
+								});
+							}
+							else {
+								match = true;
+							}
+
+							if (match) {
+
+								// get the image keyed as Normal in the API response
+								const imageData = item.multimedia.find((element) => {
+									return element.format === 'Normal';
+								});
+
+								// pass along relevant data for results
+								const matchedArticle = {
+									title: item.title,
+									url: item.url,
+									byline: item.byline,
+									relevance: relevance,
+									image: imageData !== undefined ? imageData.url : 'https://via.placeholder.com/300',
+								};
+
+								matchedResults.push(matchedArticle);
+							}
+
+							if (Array.isArray(matchedResults) && matchedResults.length > 0) {
```

Comment:
I'm a fan of `isArray` but is there a circumstance where `matchedResults` wouldn't be an array, since it's defined as such on line 120?

Reply:
That's true, probably can just use the 2nd part of this conditional. That was just overly defensive more than anything else haha

Reply:
overly defensive > not defensive enough